staging/src/k8s.io/code-generator/cmd/register-gen/generators/packages.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"github.com/golang/glog"
+
+	clientgentypes "k8s.io/code-generator/cmd/client-gen/types"
+	"k8s.io/gengo/args"
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/namer"
+	"k8s.io/gengo/types"
+	"strings"
+)
+
+func NameSystems() namer.NameSystems {
+	return namer.NameSystems{}
+}
+
+func DefaultNameSystem() string {
+	return "public"
+}
+
+func Packages(context *generator.Context, arguments *args.GeneratorArgs) generator.Packages {
+	boilerplate, err := arguments.LoadGoBoilerplate()
+	if err != nil {
+		glog.Fatalf("Failed loading boilerplate: %v", err)
+	}
+
+	packages := generator.Packages{}
+	for _, inputDir := range arguments.InputDirs {
```

Comment:
maybe you do that elsewhere already: if there is a register.go, skip the generation.

Reply:
> maybe you do that elsewhere already: if there is a register.go, skip the generation.

Skip the generation if register.go is already there, but I expected this file to be named something like `generated.register.go`.  I figure some other poor soul will try to plumb it into our build scripts.

Reply:
more like zz_generated.register.go, like the other generated files.

Reply:
do we want to skip the generation if there is `register.go` or `zz_generated.register.go` ? I think that we have two cases here:

* a user already has `register.go` fille and runs the new version of the generators. If we will generate `zz_generated.register.go` this might be not only confusing but also can cause compilation errors.

* a user changes `types.go` file and runs generators. It would be nice if we could reflect the changes in `zz_generated.register.go` otherwise the user would have to manually delete the file.

Reply:
If `register.go` exists, just continue with the next file.
If `zz_generated.register.go` exists, override it.